### PR TITLE
Added 4.x suite for Interop testing 

### PR DESCRIPTION
**Pull request by shuchitajnn**
# Description

Added nautilius suite for 4.x interop testing.

Logs:
[interop_rpm_logs.zip](https://github.com/red-hat-storage/cephci/files/8471848/interop_rpm_logs.zip)
[interop_container_logs.zip](https://github.com/red-hat-storage/cephci/files/8471849/interop_container_logs.zip)

### CODE REVIEW

File: suites/nautilus/interop/test_ceph_sanity_container.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            mon:
+              mon_allow_pool_delete: true
+          ceph_docker_image: rhceph/rhceph-4-rhel8
+          ceph_docker_image_tag: latest
+          ceph_docker_registry: registry.redhat.io
+          ceph_origin: distro
+          ceph_repository: rhcs
+          ceph_stable: true
+          ceph_stable_release: nautilus
+          ceph_stable_rh_storage: true
```
Comment (psathyan) [inline suites/nautilus/interop/test_ceph_sanity_container.yaml:35]: Our intention is to use the released versions of RHCS, hence we may want to change this to
```
ceph_origin: repository
ceph_repository: rhcs
ceph_repository_type: cdn
ceph_rhcs_version: 4
```

File: suites/nautilus/interop/test_ceph_sanity_container.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+#-----    Sanity Test suite     ------
+#-------------------------------------
+#===============================================================================================
+tests:
+  - test:
+      abort-on-fail: true
+      module: install_prereq.py
+      name: "install ceph pre-requisites"
+  - test:
+      abort-on-fail: true
```
Comment (psathyan) [inline suites/nautilus/interop/test_ceph_sanity_container.yaml:12]: You may want to to add `use_cdn: true`

File: suites/nautilus/interop/test_ceph_sanity_container.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          ceph_stable_release: nautilus
+          ceph_stable_rh_storage: true
+          ceph_test: true
+          cephfs_pools:
+            -
+              name: cephfs_data
+              pgs: "8"
+            -
+              name: cephfs_metadata
+              pgs: "8"
```
Comment (psathyan) [inline suites/nautilus/interop/test_ceph_sanity_container.yaml:43]: Not sure if these settings are required.

File: suites/nautilus/interop/test_ceph_sanity_container.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            desc: Executig upstream RBD CLI Generic scenarios
+            module: test_rbd.py
+            name: 1_rbd_cli_generic
+            polarion-id: CEPH-83574241
+        - test:
+            name: cephfs-basics
+            module: cephfs_basic_tests.py
+            polarion-id: CEPH-11293,CEPH-11296,CEPH-11297,CEPH-11295
+            desc: cephfs basic operations
+            abort-on-fail: false
```
Comment (psathyan) [inline suites/nautilus/interop/test_ceph_sanity_container.yaml:92]: IMO, it is sufficient to check if the deployment using containers is good as the `rpm` based testing would cover the functionality.

File: suites/nautilus/interop/test_ceph_sanity_rpm.yaml [hunk 5]
```diff
@@ -0,0 +1,88 @@
+#===============================================================================================
```
Comment (psathyan) [inline suites/nautilus/interop/test_ceph_sanity_rpm.yaml:1]: Likewise the above comments apply to this file also... that is

- rename the file to `test-ceph-rpms.yaml`
- check the test configurations used for deployment
- check the ansi_config

File: suites/nautilus/interop/test_ceph_sanity_container.yaml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+          ceph_conf_overrides:
+            client:
+              rgw crypt require ssl: false
+              rgw crypt s3 kms encryption keys: testkey-1=YmluCmJvb3N0CmJvb3N0LWJ1aWxkCmNlcGguY29uZgo=
+                testkey-2=aWIKTWFrZWZpbGUKbWFuCm91dApzcmMKVGVzdGluZwo=
+            global:
+              mon_max_pg_per_osd: 1024
+              osd_default_pool_size: 2
+              osd_pool_default_pg_num: 64
+              osd_pool_default_pgp_num: 64
```
[the remark this answers is not in the document]
Comment (shuchitajnn) [reply suites/nautilus/interop/test_ceph_sanity_container.yaml:25]: Okay @psathyan, so should I remove them for now?

[the remark this answers is not in the document]
Comment (psathyan) [reply suites/nautilus/interop/test_ceph_sanity_container.yaml:25]: IMO, we can remove it

@tintumathew10 what are your thoughts?

**Comment by tintumathew10** [comment]
another PR was created and merged to master.

**Pull request closed without merging**